### PR TITLE
[Fix] single pass input view

### PR DESCRIPTION
**Pull request by rrahn**
Fixes construction of single pass input view.

### CODE REVIEW

File: include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp [hunk 1]
```diff
@@ -43,7 +43,8 @@ template <std::ranges::ViewableRange resource_t,
           typename alignment_algorithm_t,
           typename execution_handler_t = execution_handler_sequential>
 //!\cond
-    requires std::ranges::InputRange<std::remove_reference_t<resource_t>>
+    requires std::ranges::InputRange<std::remove_reference_t<resource_t>> &&
```
Comment (h-2) [inline include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp]: don't remove reference

File: include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp [hunk 2]
```diff
@@ -96,8 +97,8 @@ class alignment_executor_two_way
     //!\brief Constructs this executor with the passed range of alignment instances.
     alignment_executor_two_way(resource_t && resrc,
                                alignment_algorithm_t fn) :
-        resource{view::single_pass_input(std::forward<resource_t>(resrc))},
-        kernel{fn}
+        resource{static_cast<resource_t &&>(resrc)},
```
Comment (h-2) [inline include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp]: this is literally the same as `std::forward`

File: include/seqan3/range/view/single_pass_input.hpp [hunk 3]
```diff
@@ -83,16 +81,33 @@ class single_pass_input_view : public detail::view_base
...
+    {}
+
+    //!\brief Construction from std::ranges::ViewableRange.
+    template <typename other_urng_t>
+    //!\cond
+    requires !std::Same<remove_cvref_t<other_urng_t>, single_pass_input_view> &&
+             std::ranges::ViewableRange<other_urng_t> &&  // Must come after self type check to avoid conflicts with the move constructor.
+             std::Constructible<urng_t, ranges::ref_view<std::remove_reference_t<other_urng_t>>>
+    //!\endcond
+    explicit single_pass_input_view(other_urng_t && _urng) :
```
Comment (h-2) [inline include/seqan3/range/view/single_pass_input.hpp]: we need to decide on the explicitness of the constructors

File: include/seqan3/range/view/single_pass_input.hpp [hunk 4]
```diff
@@ -83,16 +81,33 @@ class single_pass_input_view : public detail::view_base
...
+        single_pass_input_view{std::view::all(_urng)}
     {}
```
Comment (h-2) [inline include/seqan3/range/view/single_pass_input.hpp]: How do you set the state of a default-constructed single_pass view with these constructors?

File: include/seqan3/range/view/single_pass_input.hpp [hunk 5]
```diff
@@ -137,9 +150,14 @@ class single_pass_input_view : public detail::view_base
  * \relates seqan3::detail::single_pass_input_view
  * \{
  */
-//!\brief Deduces the single_pass_input_view from the underlying range.
-template <std::ranges::InputRange urng_t>
-single_pass_input_view(urng_t &&) -> single_pass_input_view<urng_t>;
+
+//!\brief Deduces the single_pass_input_view from the underlying range if it is a std::ranges::ViewableRange.
+template <std::ranges::View urng_t>
+single_pass_input_view(urng_t) -> single_pass_input_view<urng_t>;
```
Comment (h-2) [inline include/seqan3/range/view/single_pass_input.hpp]: this is superfluous

File: include/seqan3/range/view/single_pass_input.hpp [hunk 6]
```diff
@@ -137,9 +150,14 @@ class single_pass_input_view : public detail::view_base
...
+//!\brief Deduces the single_pass_input_view from the underlying range if it is a std::ranges::ViewableRange.
+template <std::ranges::ViewableRange urng_t>
+single_pass_input_view(urng_t &&) ->
+    single_pass_input_view<decltype(std::view::all(std::declval<urng_t>()))>;
```
Comment (h-2) [inline include/seqan3/range/view/single_pass_input.hpp]: we have `std::ranges::all_view< >`

File: include/seqan3/range/view/single_pass_input.hpp (hunk 4 above)
Comment (rrahn) [reply include/seqan3/range/view/single_pass_input.hpp]: As discussed. The pointer is reset with the new allocated state.

File: include/seqan3/range/view/single_pass_input.hpp (hunk 3 above)
Comment (rrahn) [reply include/seqan3/range/view/single_pass_input.hpp]: Well, yes. I am strongly in favor of this.

**Pull request merged**